### PR TITLE
Fix Options Source Gen with Length attributes applied on properties of Interface type

### PR DESCRIPTION
**Pull request by tarekgh**
https://github.com/dotnet/runtime/issues/93403

This is fixing a regression from last changes https://github.com/dotnet/runtime/pull/93088. The problem was when we were looking for the `Count` property inside a type, if the type is interfacing the lookup was failing. The reason is `ITypeSymbol.GetMembers` doesn't report the members of the extended interfaces. The fix is just lookup the property in all interfaces implemented by this type.

### CODE REVIEW

**Review by ericstj — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Baselines/UsingInterfaceAsPropertyTypeForLengthAttributesTests.netcore.g.cs [hunk 1]
```diff
@@ -0,0 +1,252 @@
+
+    // <auto-generated/>
+    #nullable enable
+    #pragma warning disable CS1591 // Compensate for https://github.com/dotnet/roslyn/issues/54103
+    namespace Test
+{
```
Comment (RussKie) [inline src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Baselines/UsingInterfaceAsPropertyTypeForLengthAttributesTests.netcore.g.cs:6]: [nit] mis-indented

File: src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Baselines/UsingInterfaceAsPropertyTypeForLengthAttributesTests.netfx.g.cs [hunk 2]
```diff
@@ -0,0 +1,177 @@
+
+    // <auto-generated/>
+    #nullable enable
+    #pragma warning disable CS1591 // Compensate for https://github.com/dotnet/roslyn/issues/54103
+    namespace Test
```
Comment (RussKie) [inline src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Baselines/UsingInterfaceAsPropertyTypeForLengthAttributesTests.netfx.g.cs:5]: ditto

File: src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Baselines/UsingInterfaceAsPropertyTypeForLengthAttributesTests.netcore.g.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Options/tests/SourceGeneration.Unit.Tests/Baselines/UsingInterfaceAsPropertyTypeForLengthAttributesTests.netcore.g.cs:6]: We can fix this in another PR. I want to limit the change here as we need to port this to 8.0. Thanks for the comment.

**Comment by tarekgh** [comment]
failures are tracked by https://github.com/dotnet/runtime/issues/92423 which are unrelated. I am not sure why the build analysis didn't detect it automatically. CC @agocke

**Pull request merged**